### ISSUE TITLE
Per-realm turn_secret

### ISSUE THREAD

**Issue by OrangeDog** [coturn/coturn#148]
I cannot get per-realm secrets working with Coturn-4.5.0.3. only the default realm is ever used.
```bash
turnserver -n --lt-cred-mech --listening-ip 127.0.0.1 --realm default --psql-userdb "host=192.168.56.1 dbne=postgres user=coturn password=coturn" --use-auth-secret
```
```sql
INSERT INTO turn_secret (realm, value) VALUES ('default', 'one'), ('example.com', 'two');
INSERT INTO turn_origin_to_realm (origin, realm) VALUES ('http://example.com', 'example.com');
```
The first of these tests succeeds as expected, but the second fails to allocate.
```bash
turnutils_uclient -u testuser -W one 127.0.0.1
turnutils_uclient -u testuser -W two -o 'http://example.com' 127.0.0.1
```
```
189: ERROR: check_stun_auth: Cannot find credentials of user <1496496862:testuser>
```
Am I doing something wrong?
If it's not supposed to work, why is there a realm column in the turn_secret table?

**Comment by mom040267** [comment]
It does work. Run the example server scripts, and use a client script with
option -J. Then you will see how it works.

But the problem is that the realm specs have not succeeded to become an
RFC, and we do not recommend using them anymore. Eventually, the realms
will be removed from coturn.

**Comment by OrangeDog** [comment]
> -J Use oAuth with default test key kid='north'.

I'm not talking about oAuth. This is the "TURN REST API" secret.

**Comment by mom040267** [comment]
REST API examples are in examples/scripts/restapi/

As I said, using realms is not recommended in the long-term

**Comment by OrangeDog** [comment]
Yes I know where the examples are, and there isn't one that uses multiple realms.

And I know you said that, but you said it was because realms didn't succeed to become an RFC, yet they are clearly in the core STUN specification.

**Comment by mom040267** [comment]
single realm is in the core STN authentication, I thought you are talking
about multiple realms. Single realm is supported, or course. Do not mix
those things.

**Comment by OrangeDog** [comment]
What are you talking about?
Of course you can only authenticate with one realm at a time, but it is expected for the server to support multiple realms, otherwise there's no point in telling the client which realm to use.
Furthermore, the structure of the database tables are clearly designed to support per-realm users, secrets, parameters, etc. with an origin->realm mapping.
So why does it not work?

**Comment by mom040267** [comment]
Multiple realms do work with usual long term authentication. They also work with oauth. We do not support REST with multiple realms.

**Comment by OrangeDog** [comment]
Then why have you added a realm column to the auth_secret table?
Either remove it, or actually support it.

**Comment by SierraAlphaWhiskey** [comment]
Could we get some clarification on this? When configuring applications such as bareSIP or LINPhone, I need to specify a user/password combination for TURN (lt-cred-mech). For Prosody, it uses the dynamic, time-based method with a shared secret (use-auth-secret). Furthermore, ICE testing sites such as [Trickle ICE](https://webrtc.github.io/samples/src/content/peerconnection/trickle-ice/) and [IceTest.info](https://icetest.info/?) seem to only work with lt-cred-mech. I have spent quite some time reading the docs, examples, and source trying to make these work together. The example script, contrary to the docs, sets both lt-cred-mech and use-auth-secret. The docs report these are incompatible. With both specified, I even get a warning:
> WARNING: CONFIG: You specified --lt-cred-mech and --use-auth-secret in the same time.
Be aware that you could not mix the username/password and the shared secret based auth methods. 
Shared secret overrides username/password based auth method. Check your configuration!

So, with all that out of the way and reading what has been written above, is there a way to allow both authentication mechanisms to work? The example script even says:
```
# This is an example how to start a TURN Server in
# secure 'dynamic' 'secret' mode (see https://github.com/coturn/coturn/blob/master/README.turnserver)
# with SQLite database for users information
# with the long-term credentials mechanism.
```
That is EXACTLY what I want to do. So far, my best guess at a config is:
```
#lt-cred-mech
use-auth-secret
#static-auth-secret=secret
```
This works for Prosody using the time-based method and shared secret from the turn_secret table. It does not work for LINPhone or the ICE test pages. In reading what mom040267 said:
> ... We do not support REST with multiple realms.
Does that mean what I am trying to do is not supported?
